### PR TITLE
Added support for HBM backward compartibility of existing application

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
Added support for HBM backward compartibility of existing application

### CODE REVIEW

**Comment by chvamshi-xilinx** [comment]
@saifuddin-xilinx , As discussed offline, Please simplify the code and dont remove the DRC.

**Review by chvamshi-xilinx — APPROVED** [review]
Looks good.  Can you please add a comment why DRC checks are performed before adding origincal connections.  It will be helpful

File: src/runtime_src/xocl/core/device.cpp [hunk 1]
```diff
@@ -713,7 +713,7 @@ get_cu_memidx() const
         mask &= cu->get_memidx_intersect();
 
       // select first common memory bank index if any
-      for (size_t idx=0; idx<mask.size(); ++idx) {
+      for (size_t idx=mask.size() - 1; idx>=0; --idx) {
```
Comment (stsoe) [inline src/runtime_src/xocl/core/device.cpp:716]: Please elaborate on this change.  The comment no longer reflects the loop.  Why was it changed?

Comment (stsoe) [reply src/runtime_src/xocl/core/device.cpp]: The OCL code should be using GROUP_CONNECTIVITY and the connectivity in this section should always be valid groups.

@saifuddin-xilinx explains that APPs requested backwards compatibility to old mem topology indices, but we have to revisit this request.  Grouping is a new feature, existing host code and xclbins will continue to work without any changes.  

However changing the xclbin to use grouping of HBMs and changing CU connectivity implies that host code must change also.   One should not expect existing host code to work with xclbins that have been converted to use groups.

We should resolve this and revert the change in xocl/core/device.cpp.

**Comment by chvamshi-xilinx** [comment]
Having offline discussions with @stsoe on this page.
https://confluence.xilinx.com/display/XSW/Specifying+Bank+index+in+Application

**Comment by saifuddin-xilinx** [comment]
This is not required for now. We will see if there is any real use case..

**Pull request closed without merging**